components/places/src/storage/history_metadata.rs

```diff
@@ -1297,6 +1315,102 @@ mod tests {
         assert_eq!(0, get_since(&conn, after_meta2).unwrap().len());
     }
 
+    #[test]
+    fn test_get_most_recent_empty() {
+        let conn = PlacesDb::open_in_memory(ConnectionType::ReadWrite).expect("memory db");
+        let rows = get_most_recent(&conn, 5).expect("query ok");
+        assert!(rows.is_empty());
+    }
+
+    #[test]
+    fn test_get_most_recent_orders_and_limits() {
+        let conn = PlacesDb::open_in_memory(ConnectionType::ReadWrite).expect("memory db");
+
+        note_observation!(&conn,
+            url "https://example.com/1",
+            view_time Some(10),
```

Comment:
we should probably have this identical for all visits to make it clear it doesn't matter. Related, this might be racey in that it probably assumes incrementing timestamps between observations, which might not be true, so the order back is different. I'm not immediately sure what to do about that :( Maybe ignore it if other tests rely on it, sqlite doing a full transaction before the timestamp advances seems unlikely to be a problem for a while?

Reply:
> Related, this might be racey in that it probably assumes incrementing timestamps between observations, 
Yeah I can see how that can be a problem. I don't see anything in the other tests that attempts to solve it other than maybe some `thread::sleep` that might have been added for this ?

Reply:
Sweet ! Done in https://github.com/mozilla/application-services/pull/7002/commits/f8acb29202c6e93b7ca5f15f9cb9ac4ec268e228. I added a tiny helper with a comment. @mhammond if it looks good can you please merge after CI is green. I don't have merge access. Thanks !